source/_components/image_processing.tensorflow.markdown

```diff
@@ -0,0 +1,161 @@
+---
+layout: page
+title: "TensorFlow"
+description: "Detect and recognize objects with TensorFlow."
+date: 2018-10-24 00:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: tensorflow.png
+ha_category: Image Processing
+featured: false
+ha_release: 0.82
+---
+
+The `tensorflow` image processing platform allows you to detect and recognize objects in a camera image using [TensorFlow](https://www.tensorflow.org/). The state of the entity is the number of objects detected, and recognized objects are listed in the `summary` attribute along with quantity. The `matches` attribute provides the confidence `score` for recognition and the bounding `box` of the object for each detection category.
+
+<p class='note warning'>
+  This package will not yet work on Hass.io installations, but should work on Hassbian.  [See here](https://www.tensorflow.org/install/pip) for system requirements.
+</p>
+
+## {% linkable_title Setup %}
+
+TensorFlow object detection requires some dependencies outside of the core tensorflow package.  These live in the [tensorflow/models](https://github.com/tensorflow/models/tree/master/research/object_detection) GitHub repository.  The one we are interested in for this use case lives in `research/object_detection`.  A general outline of the installation process is as follows:
+
+- Clone [tensorflow/models](https://github.com/tensorflow/models/tree/master/research/object_detection)
+- Compile protobuf models located in `research/object_detection/protos` with `protoc`
+- Create the following directory structure inside your config directory:
+
+```bash
+  |- {config_dir}
+    | - tensorflow
+      |- __init__.py
+      |- object_detection/
+```
+
+- Copy required object_detection dependancies to the `object_detection` folder inside of the `tensorflow` folder:
+    - `research/object_detection/data`
+    - `research/object_detection/utils`
+    - `research/object_detection/protos`
+
+A sample script can be found at [this gist](https://gist.github.com/hunterjm/6f9332f92b60c3d5e448ad936d7353c3).
+
+## {% linkable_title Model Selection %}
+
+Lastly, it is time to pick a model.  It is recommended to start with one of the COCO models available in the [Model Detection Zoo](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md).
+
+The trade-off between the different models is accuracy vs speed.  Users with a decent CPU should start with the `faster_rcnn_inception_v2_coco` model.  If you are running on an ARM device like a Rasberry Pi, start with the `ssd_mobilenet_v2_coco` model.
+
+Whichever model you choose, download it and place the `frozen_inference_graph.pb` file somewhere in your config directory.
```

Comment:
Perhaps its worth recommending somewhere specific rather than `somewhere`, perhaps in the `tensorflow` dir we created earlier?